Source/SwiftLintFramework/Rules/Style/ImplicitGetterRule.swift

```diff
@@ -16,134 +15,63 @@ public struct ImplicitGetterRule: ConfigurationProviderRule {
     )
 
     public func validate(file: SwiftLintFile) -> [StyleViolation] {
-        let getTokens = findGetTokens(file: file)
-
-        let violatingLocations = getTokens.compactMap { token -> (ByteCount, SwiftDeclarationKind?)? in
-            // the last element is the deepest structure
-            guard let dict = declarations(forByteOffset: token.offset,
-                                          structureDictionary: file.structureDictionary).last else {
-                return nil
-            }
-
-            // If there's a setter, `get` is allowed
-            if SwiftVersion.current < .fiveDotTwo || dict.accessibility != nil {
-                guard dict.setterAccessibility == nil else {
-                    return nil
-                }
-            } else {
-                guard let range = dict.byteRange.flatMap(file.stringView.byteRangeToNSRange) else {
-                    return nil
-                }
-
-                let setTokens = findSetTokens(file: file, range: range)
-                let hasSetToken = setTokens.contains { token in
-                    // the last element is the deepest structure
-                    guard let setDict = declarations(forByteOffset: token.offset,
-                                                     structureDictionary: file.structureDictionary).last else {
-                        return false
-                    }
-
-                    return setDict.offset == dict.offset
-                }
-
-                guard !hasSetToken else {
-                    return nil
-                }
+        let visitor = ImplicitGetterRuleVisitor()
+
+        return visitor
```

Comment:
Nit: I like using implicit returns for these

```suggestion
        ImplicitGetterRuleVisitor()
```